chirp/drivers/tdh8.py

```diff
@@ -2062,3 +2477,53 @@ def validate_memory(self, mem):
 class UV68(TDH8):
     VENDOR = "TID"
     MODEL = "TD-UV68"
+
+
+@directory.register
+class TDH3(TDH8):
+    VENDOR = "TIDRADIO"
+    MODEL = "TD-H3"
+    _memsize = 0x1fef
+    _ranges_main = [(0x0000, 0x1fef)]
+    _idents = [TD_H3]
+    _vhf_range = (136000000, 175000000)
+    _uhf_range = (400000000, 521000000)
+    _aux_block = True
+    _tri_power = True
+    _gmrs = False
+    _ham = False
+    _mem_params = (0x1F2F)
+
+
+@directory.register
+class TDH3_HAM(TDH3):
+    VENDOR = "TIDRADIO"
+    MODEL = "TD-H3-HAM"
+    ident_mode = b'P31185\xff\xff'
+    _ham = True
+    _vhf_range = (144000000, 149000000)
+    _uhf_range = (420000000, 451000000)
+
+
+@directory.register
+class TDH3_GMRS(TDH3):
+    VENDOR = "TIDRADIO"
+    MODEL = "TD-H3-GMRS"
+    ident_mode = b'P31184\xff\xff'
+    _gmrs = True
+    _vhf_range = (136000000, 175000000)
+    _uhf_range = (400000000, 521000000)
+
+    def validate_memory(self, mem):
+        msgs = super().validate_memory(mem)
+        if 31 <= mem.number <= 54 and mem.freq not in GMRS_FREQS:
+            msgs.append(chirp_common.ValidationError(
+                "The frequency in channels 31-54 must be between"
+                "462.55000-462.72500 in 0.025 increments."))
+        return msgs
+
+
+@directory.register
+class TIDH3(TDH3):
+    VENDOR = "TID"
+    MODEL = "TD-H3"
```

Comment:
This adds a clone of the radio under a slightly-different vendor. Please do not do that and remove this last subclass.